transformations/transformations/data_offload.py

```diff
@@ -134,10 +138,16 @@ def insert_data_offload_pragmas(self, routine, targets):
                 inoutargs = tuple(dict.fromkeys(inoutargs))
 
                 # Now geenerate the pre- and post pragmas (OpenACC)
-                copyin = f'copyin({", ".join(inargs)})' if inargs else ''
-                copy = f'copy({", ".join(inoutargs)})' if inoutargs else ''
-                copyout = f'copyout({", ".join(outargs)})' if outargs else ''
-                pragma = Pragma(keyword='acc', content=f'data {copyin} {copy} {copyout}')
+                if self.deviceptr:
+                    deviceptr = ''
+                    if inargs+outargs+inoutargs:
+                        deviceptr = f'deviceptr({", ".join(inargs+outargs+inoutargs)})'
+                    pragma = Pragma(keyword='acc', content=f'data {deviceptr}')
```

Comment:
```suggestion
                if self.deviceptr:
                    offload_args = inargs + outargs + inoutargs
                    if offload_args:
                        deviceptr = f' deviceptr({", ".join(offload_args)})'
                    else:
                        deviceptr = ''
                    pragma = Pragma(keyword='acc', content=f'data{deviceptr}')
```

Purely a matter of taste, but I find this a little hard to read with the repeated contracted "sum".

Reply:
Seconded (but yes, matter of taste), spaces around operators used to be a pylint rule (for a reason).

Reply:
Seconded (but yes, matter of taste), spaces around operators used to be a pylint rule (for a reason).